packages/jest-types/src/Global.ts

```diff
@@ -106,10 +106,15 @@ export interface HookBase {
   (fn: HookFn, timeout?: number): void;
 }
 
+export interface Failing {
+  (testName?: TestNameLike, fn?: TestFn, timeout?: number): void | never;
```

Comment:
Keep in mind these types will be consumed through `import {test} from '@jest/globals'`. Hence `testName` or `fn` cannot be optional.

Would be very good to add type tests for new APIs. Relevant ones live in [`packages/jest-types/__typetests__/globals.test.ts`](https://github.com/facebook/jest/blob/main/packages/jest-types/__typetests__/globals.test.ts) and [`packages/jest-types/__typetests__/each.test.ts`](https://github.com/facebook/jest/blob/main/packages/jest-types/__typetests__/each.test.ts)

Tests of `each` are rather detailed. I think in this case it should be enough to do something like here: https://github.com/facebook/jest/blob/a5b52a520591bcd5500538afea94b6994d9fd909/packages/jest-types/__typetests__/each.test.ts#L445-L446

Ah.. To run the type tests locally: build the library and run `yarn test-types`

Reply:
Makes sense, I will add these as well.

Reply:
made these types as required, do you still reckon some tests need to be added?

Reply:
It would be useful to add a type test for `test.failing.each` at the bottom of `test.failing` tests: 

https://github.com/facebook/jest/blob/a5b52a520591bcd5500538afea94b6994d9fd909/packages/jest-types/__typetests__/globals.test.ts#L295-L297

Just a single line, similar to `test.only.each`: https://github.com/facebook/jest/blob/a5b52a520591bcd5500538afea94b6994d9fd909/packages/jest-types/__typetests__/globals.test.ts#L317

Reply:
Oh.. I looked at documentation only and saw just this one case. There are more, right? `test.concurrent.failing.each`, `test.concurrent.only.failing.each`, `test.concurrent.skip.failing.each`, `test.only.failing.each`, `test.skip.failing.each` must be also type tested. Anything else? I think these should be documented as well.

The type tests help to prevent regression in user-facing types in the future. This library is huge. Easy to break something without noticing.

Reply:
Got it, will add these tests as well 👍🏻

Reply:
Just this left I think, and we're good to go! 😀

Reply:
Added more [tests](https://github.com/facebook/jest/pull/13142/commits/03a3b31ec548deb9100ee453604eec2a665bb738)